### PR TITLE
allow specifying Authorization header in startWebRequestSync

### PR DESCRIPTION
**Pull request by davidsbailey**
This allows the `Authorization` header to be specified via the startWebRequestSync block and passed through the xhr proxy to the remote endpoint. This enables sending web requests for endpoints such as Spotify and Twitter. The startWebRequestSync block is undocumented and currently used only for prototyping.

### CODE REVIEW

**Comment by davidsbailey** [comment]
sample applab code:
```js
var headers = {
  Authorization: 'Bearer abcdefg...',
};
var result = startWebRequestSync(
  'https://api.spotify.com/v1/browse/new-releases', 
  headers
);
console.log(result);
```
see documentation for this API [here](https://developer.spotify.com/console/get-new-releases).

File: apps/src/applab/api.js [hunk 1]
```diff
@@ -380,8 +380,8 @@ export function startWebRequest(url, func) {
                            'func': func });
 }
 
-export function startWebRequestSync(url, func) {
-  return Applab.executeCmd(null, 'startWebRequestSync', {url, func});
+export function startWebRequestSync(url, func, headers) {
```
Comment (joshlory) [inline apps/src/applab/api.js]: Have you tested this out? I think `func` might need to be last in order for native-is-async to work.

Comment (davidsbailey) [reply apps/src/applab/api.js]: Great catch Josh! This was in fact broken and is fixed now.

I initially had it working with `func` last, swapped the param order, and then must not have properly retested.

**Comment by davidsbailey** [comment]
@joshlory , please take another look!

Here is a screenshot:
screen shot 2018-12-17 at 1 28 00 pm

**Comment by davidsbailey** [comment]
also, heads up @gtw that the spotify access tokens expire after a short while (20 minutes? 2 hours?) so for now you have to keep getting a new one for the app to keep working. It will probably need to be this way for piloting, however there is probably a way to eliminate the need for this in the long run.

**Review by joshlory — APPROVED** [review]
Nice, LGTM!

**Pull request merged**